edweiss-app/components/lectures/remotecontrol/modal.tsx

```diff
@@ -519,3 +520,116 @@ export const QuestionBroadcastModal: ReactComponent<{
         </ModalContainer>
     );
 };
+
+
+
+// ------------------------------------------------------------
+// ---------------      Quiz Broadcast Modal     --------------
+// ------------------------------------------------------------
+
+export const QuizBroadcastModal: ReactComponent<{
+    modalRef: React.RefObject<BottomSheetModalMethods>;
+    id: string;
+    quizModel?: LectureQuizzes.LectureQuiz;
+    courseId: string,
+    lectureId: string,
+    broadcasted: string,
+    setBroadcasted: React.Dispatch<React.SetStateAction<string>>;
+    onClose: () => void;
+}> = ({ modalRef, id, courseId, lectureId, quizModel, broadcasted, setBroadcasted, onClose }) => {
+
+
+    // Hooks
+    const [broadLoading, setBroadLoading] = useState(false);
+    const [showLoading, setShowLoading] = useState(false);
+    const [showResultToStudent, setShowResultToStudent] = useState(false);
+
+    useEffect(() => {
+        if (quizModel?.showResultToStudents) {
+            setShowResultToStudent(quizModel.showResultToStudents);
+        }
+    }, [quizModel?.showResultToStudents]);
+
+
+    async function toggleResult() {
+        setShowLoading(true);
+        const res = await callFunction(Quizzes.Functions.toggleLectureQuizResult, { lectureId: lectureId, lectureEventId: id, courseId: courseId, });
+
+        if (res.status === 1) {
+            console.log(`toggled showResultToStudent boolean`);
+        } else {
+            console.log(`Error while toggling boolean shoresultToStudent`);
+        }
+        setShowResultToStudent(!showResultToStudent)
+        setShowLoading(false);
+    }
+
+
+
+    async function handleQuizBroadcast() {
+        if (broadcasted === id) {
+
+            setBroadLoading(true);
+
+            // End of quiz activity
+            try {
```

Comment:
What kind of thrown exception you expect from the `callFunction` ?
You are not handling the errors as they should, please store the `callFunction` returns and check if their `status == 0`.